<?xml version="1.0"?>
<launch>

    <include file="$(find akros_test)/launch/akros_description.launch" />
    <include file="$(find depthai_nodes)/launch/stereo_publisher.launch" />
    <include file="$(find rplidar_ros)/launch/rplidar_publisher.launch" /> 
    
    <arg name="nodeName"         default="stereo_pub"/>
    <arg name="camera_name"      default="oakd" />
    <arg name="camera_param_uri" default="package://depthai_nodes/params/camera" />
    
    <!-- depth_image_proc nodelet -->
    <node pkg="nodelet" type="nodelet" name="pointcloud_manager" args="manager" output="screen"/>
    
    <!-- <node pkg="nodelet" type="nodelet" name="pointcloud_manager"  args="manager" output="screen">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>
    </node> -->

    <!-- node pkg="nodelet" type="nodelet" name="depth_convert"
        args="load depth_image_proc/convert_metric pointcloud_manager">
        <remap from="image_raw" to="/$(arg nodeName)/stereo/depth"/>    
        <remap from="image" to="/$(arg nodeName)/stereo/depth_m"/>
    </node -->

    <node pkg="nodelet" type="nodelet" name="depth_to_pointcloud"
        args="load depth_image_proc/point_cloud_xyz pointcloud_manager">
        <remap from="camera_info" to="/$(arg nodeName)/stereo/camera_info"/>
        <remap from="image_rect" to="/$(arg nodeName)/stereo/depth"/>
        <remap from="points" to="/$(arg nodeName)/stereo/points"/>
    </node>

</launch>
